openedx/core/djangoapps/api_admin/models.py

```diff
@@ -88,3 +97,75 @@ class ApiAccessConfig(ConfigurationModel):
 
     def __unicode__(self):
         return u'ApiAccessConfig [enabled={}]'.format(self.enabled)
+
+
+@receiver(post_save, sender=ApiAccessRequest, dispatch_uid="api_access_request_post_save_email")
+def send_request_email(sender, instance, created, **kwargs):   # pylint: disable=unused-argument
+    """ Send request email after new record created. """
+    if created:
+        _send_new_pending_email(instance)
+
+
+@receiver(pre_save, sender=ApiAccessRequest, dispatch_uid="api_access_request_pre_save_email")
+def send_decision_email(sender, instance, **kwargs):  # pylint: disable=unused-argument
+    """ Send decision email after status changed. """
+    if instance.id and not instance.contacted:
+        old_instance = ApiAccessRequest.objects.get(pk=instance.id)
+        if instance.status != old_instance.status:
+            _send_decision_email(instance)
+
+
+def _send_new_pending_email(instance):
+    """ Send an email to settings.API_ACCESS_MANAGER_EMAIL with the contents of this API access request. """
+    context = {
+        'approval_url': ''.join([
+            instance.base_url,
+            reverse('admin:api_admin_apiaccessrequest_change', args=(instance.id,))
+        ]),
+        'company_name': instance.company_name,
+        'username': instance.user.username,
+        'url': instance.website,
+        'company_address': instance.company_address,
+        'reason': instance.reason,
+    }
+
+    message = render_to_string('api_admin/api_access_request_email_new_request.txt', context)
+    try:
+        send_mail(
+            _('API access request from {company}').format(company=instance.company_name),
+            message,
+            settings.API_ACCESS_FROM_EMAIL,
+            [settings.API_ACCESS_MANAGER_EMAIL],
+            fail_silently=False
+        )
+    except SMTPException:
+        log.exception('Error sending API request email for request from [%s].', instance.user.username)
+
+
+def _send_decision_email(instance):
+    """ Send an email to requesting user with the decision made about their request. """
+    context = {
+        'name': instance.user.username,
+        'api_management_url': ''.join([
```

Comment:
You can use [`urljoin`](https://docs.python.org/2/library/urlparse.html#urlparse.urljoin) here.